src/infrastructure/backends/claude.rs

```diff
@@ -0,0 +1,240 @@
+struct Model {
+    id: String,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+struct ModelListResponse {
+    data: Vec<Model>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+struct MessageRequest {
+    role: String,
+    content: String,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+struct CompletionRequest {
+    model: String,
+    max_tokens: u32,
+    messages: Vec<MessageRequest>,
+    stream: bool,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+struct Healthcheck {
+    message: String,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+struct CompletionDeltaResponse {
+    #[serde(rename = "type")]
+    _type: String,
+    text: String,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+struct CompletionResponse {
+    #[serde(rename = "type")]
+    _type: String,
+    delta: CompletionDeltaResponse,
+}
+
+pub struct Claude {
+    url: String,
+    token: String,
+    timeout: String,
+}
+
+impl Default for Claude {
+    fn default() -> Claude {
+        return Claude {
+            url: "https://api.anthropic.com".to_string(),
+            token: Config::get(ConfigKey::ClaudeToken),
+            timeout: Config::get(ConfigKey::BackendHealthCheckTimeout),
+        };
+    }
+}
+
+#[async_trait]
+impl Backend for Claude {
+    fn name(&self) -> BackendName {
+        return BackendName::Claude;
+    }
+
+    #[allow(clippy::implicit_return)]
+    async fn health_check(&self) -> Result<()> {
+        if self.url.is_empty() {
+            bail!("Claude URL is not defined");
+        }
+        if self.token.is_empty() {
+            bail!("Claude token is not defined");
+        }
+
+        let res = reqwest::Client::new()
+            .get(format!("{url}/healthcheck", url = self.url))
+            .timeout(Duration::from_millis(self.timeout.parse::<u64>()?))
+            .send()
+            .await;
+
+        if res.is_err() {
+            tracing::error!(error = ?res.unwrap_err(), "Claude is not reachable");
+            bail!("Claude is not reachable");
+        }
+
+        let result = res.unwrap();
+        let status = result.status().as_u16();
+        if status >= 400 {
+            tracing::error!(status = status, "Claude health check failed");
+            bail!("Claude health check failed");
+        }
+
+        let json = result.json::<Healthcheck>().await?;
+        if !json.message.contains("ok") {
+            bail!("Claude health check failed");
+        }
+
+        return Ok(());
+    }
+
+    #[allow(clippy::implicit_return)]
+    async fn list_models(&self) -> Result<Vec<String>> {
+        let models: Vec<String> = vec![
+            "claude-3-sonnet-20240229".to_string(),
+            "claude-3-opus-20240229".to_string(),
+            "claude-2.1".to_string(),
+            "claude-2.0".to_string(),
+        ];
```

Comment:
🍰 I find it lazy of companies who are not making this available as an endpoint. I'm down to merge this as is, but maybe we consider scraping it from their docs page in the future. I can imagine those first two model names changing somewhat often.

https://docs.anthropic.com/claude/docs/models-overview

```javascript
[...$('.cm-s-neo')].map(e => e.innerHTML).filter((e, idx, arr) => arr.indexOf(e) === idx) 
```

Reply:
There is an issue for this already:
https://github.com/anthropics/anthropic-sdk-typescript/issues/313

I added a not so elegant solution to read the messages file from the official [Typescript SDK](https://github.com/anthropics/anthropic-sdk-typescript/blob/main/src/resources/messages.ts), then find and clean the model list with Regex. If this fails I left the backup hardcoded array.

Reply:
Smart! I love it.